### PR TITLE
* : Remove binlog related codes

### LINKED ISSUE #55949: Remove TiDB-Binlog codes.
As TiDB-Binlog has been deprecated in v8.3, we remove the code from TiDB repo according.

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?
Remove TiDB-binlog related code.

Issue Number: close #55949

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [X] No need to test
  > Only remove code. 
     

Documentation

- [X] Affects user behaviors
- [X] Contains syntax changes
- [X] Contains variable changes
- [ ] Contains experimental features
- [X] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
TiDB-Binlog related code has been deleted, since TiDB-Binlog has been deprecated.
```

### CODE REVIEW

**Comment by lance6716** [comment]
```
Please run `make bazel_prepare` to update `.bazel` files, or just apply the following git diff (run `git apply -` and paste following contents):
diff --git a/cmd/tidb-server/BUILD.bazel b/cmd/tidb-server/BUILD.bazel
index 7e498cf2ee..8a50288faf 100644
--- a/cmd/tidb-server/BUILD.bazel
+++ b/cmd/tidb-server/BUILD.bazel
@@ -27,14 +27,12 @@ go_library(
         "//pkg/server",
         "//pkg/session",
         "//pkg/session/txninfo",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/variable",
         "//pkg/statistics",
         "//pkg/store",
         "//pkg/store/copr",
         "//pkg/store/driver",
         "//pkg/store/mockstore",
-        "//pkg/tidb-binlog/pump_client",
         "//pkg/util",
         "//pkg/util/cgmon",
         "//pkg/util/chunk",
@@ -66,7 +64,6 @@ go_library(
         "@com_github_prometheus_client_golang//prometheus/push",
         "@com_github_tikv_client_go_v2//tikv",
         "@com_github_tikv_client_go_v2//txnkv/transaction",
-        "@com_github_tikv_pd_client//:client",
         "@org_uber_go_automaxprocs//maxprocs",
         "@org_uber_go_zap//:zap",
     ],
diff --git a/pkg/ddl/BUILD.bazel b/pkg/ddl/BUILD.bazel
index 786e0b66cc..3c178ab46e 100644
--- a/pkg/ddl/BUILD.bazel
+++ b/pkg/ddl/BUILD.bazel
@@ -123,7 +123,6 @@ go_library(
         "//pkg/resourcemanager/pool/workerpool",
         "//pkg/resourcemanager/util",
         "//pkg/sessionctx",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/stmtctx",
         "//pkg/sessionctx/variable",
         "//pkg/sessiontxn",
@@ -136,7 +135,6 @@ go_library(
         "//pkg/table/context",
         "//pkg/table/tables",
         "//pkg/tablecodec",
-        "//pkg/tidb-binlog/pump_client",
         "//pkg/ttl/cache",
         "//pkg/types",
         "//pkg/types/parser_driver",
diff --git a/pkg/domain/infosync/BUILD.bazel b/pkg/domain/infosync/BUILD.bazel
index 6d70e0ff9d..ce54ee1141 100644
--- a/pkg/domain/infosync/BUILD.bazel
+++ b/pkg/domain/infosync/BUILD.bazel
@@ -28,7 +28,6 @@ go_library(
         "//pkg/parser/terror",
         "//pkg/resourcegroup",
         "//pkg/session/cursor",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/variable",
         "//pkg/store/helper",
         "//pkg/tablecodec",
diff --git a/pkg/executor/BUILD.bazel b/pkg/executor/BUILD.bazel
index 895c58053b..b41069ec23 100644
--- a/pkg/executor/BUILD.bazel
+++ b/pkg/executor/BUILD.bazel
@@ -19,7 +19,6 @@ go_library(
         "brie.go",
         "brie_utils.go",
         "builder.go",
-        "change.go",
         "checksum.go",
         "compact_table.go",
         "compiler.go",
@@ -167,7 +166,6 @@ go_library(
         "//pkg/resourcemanager/util",
         "//pkg/session/txninfo",
         "//pkg/sessionctx",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/sessionstates",
         "//pkg/sessionctx/stmtctx",
         "//pkg/sessionctx/variable",
@@ -186,7 +184,6 @@ go_library(
         "//pkg/table/tables",
         "//pkg/table/temptable",
         "//pkg/tablecodec",
-        "//pkg/tidb-binlog/node",
         "//pkg/types",
         "//pkg/types/parser_driver",
         "//pkg/util",
@@ -204,7 +201,6 @@ go_library(
         "//pkg/util/deadlockhistory",
         "//pkg/util/disk",
         "//pkg/util/disttask",
-        "//pkg/util/etcd",
         "//pkg/util/execdetails",
         "//pkg/util/filter",
         "//pkg/util/format",
@@ -422,7 +418,6 @@ go_test(
         "//pkg/session",
         "//pkg/session/types",
         "//pkg/sessionctx",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/stmtctx",
         "//pkg/sessionctx/variable",
         "//pkg/sessiontxn",
diff --git a/pkg/server/handler/tests/BUILD.bazel b/pkg/server/handler/tests/BUILD.bazel
index ab9dc935e2..156ac36995 100644
--- a/pkg/server/handler/tests/BUILD.bazel
+++ b/pkg/server/handler/tests/BUILD.bazel
@@ -34,7 +34,6 @@ go_test(
         "//pkg/server/internal/util",
         "//pkg/session",
         "//pkg/sessionctx",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/stmtctx",
         "//pkg/sessionctx/variable",
         "//pkg/store/helper",
diff --git a/pkg/server/handler/tikvhandler/BUILD.bazel b/pkg/server/handler/tikvhandler/BUILD.bazel
index 852f339731..2ca42f16b2 100644
--- a/pkg/server/handler/tikvhandler/BUILD.bazel
+++ b/pkg/server/handler/tikvhandler/BUILD.bazel
@@ -22,7 +22,6 @@ go_library(
         "//pkg/session/txninfo",
         "//pkg/session/types",
         "//pkg/sessionctx",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/variable",
         "//pkg/store/gcworker",
         "//pkg/store/helper",
diff --git a/pkg/session/BUILD.bazel b/pkg/session/BUILD.bazel
index 1e3cec58a3..399fec4ab2 100644
--- a/pkg/session/BUILD.bazel
+++ b/pkg/session/BUILD.bazel
@@ -72,7 +72,6 @@ go_library(
         "//pkg/session/txninfo",
         "//pkg/session/types",
         "//pkg/sessionctx",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/sessionstates",
         "//pkg/sessionctx/stmtctx",
         "//pkg/sessionctx/variable",
diff --git a/pkg/sessionctx/variable/BUILD.bazel b/pkg/sessionctx/variable/BUILD.bazel
index 5d15e7440e..00390c7cc6 100644
--- a/pkg/sessionctx/variable/BUILD.bazel
+++ b/pkg/sessionctx/variable/BUILD.bazel
@@ -37,7 +37,6 @@ go_library(
         "//pkg/resourcegroup",
         "//pkg/sessionctx/sessionstates",
         "//pkg/sessionctx/stmtctx",
-        "//pkg/tidb-binlog/pump_client",
         "//pkg/types",
         "//pkg/types/parser_driver",
         "//pkg/util",
diff --git a/pkg/sessiontxn/isolation/BUILD.bazel b/pkg/sessiontxn/isolation/BUILD.bazel
index 56ccfa1de1..90ed949aed 100644
--- a/pkg/sessiontxn/isolation/BUILD.bazel
+++ b/pkg/sessiontxn/isolation/BUILD.bazel
@@ -22,7 +22,6 @@ go_library(
         "//pkg/planner/core",
         "//pkg/planner/core/base",
         "//pkg/sessionctx",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/variable",
         "//pkg/sessiontxn",
         "//pkg/sessiontxn/internal",
@@ -37,7 +36,6 @@ go_library(
         "@com_github_pingcap_errors//:errors",
         "@com_github_pingcap_failpoint//:failpoint",
         "@com_github_pingcap_kvproto//pkg/kvrpcpb",
-        "@com_github_pingcap_tipb//go-binlog",
         "@com_github_tikv_client_go_v2//error",
         "@com_github_tikv_client_go_v2//kv",
         "@com_github_tikv_client_go_v2//oracle",
diff --git a/pkg/store/driver/txn/BUILD.bazel b/pkg/store/driver/txn/BUILD.bazel
index 77243dd690..4e9cfcade7 100644
--- a/pkg/store/driver/txn/BUILD.bazel
+++ b/pkg/store/driver/txn/BUILD.bazel
@@ -4,7 +4,6 @@ go_library(
     name = "txn",
     srcs = [
         "batch_getter.go",
-        "binlog.go",
         "error.go",
         "scanner.go",
         "snapshot.go",
@@ -18,7 +17,6 @@ go_library(
         "//pkg/kv",
         "//pkg/meta/model",
         "//pkg/parser/mysql",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/store/driver/error",
         "//pkg/store/driver/options",
         "//pkg/table/tables",
@@ -32,7 +30,6 @@ go_library(
         "@com_github_pingcap_failpoint//:failpoint",
         "@com_github_pingcap_kvproto//pkg/kvrpcpb",
         "@com_github_pingcap_kvproto//pkg/metapb",
-        "@com_github_pingcap_tipb//go-binlog",
         "@com_github_tikv_client_go_v2//error",
         "@com_github_tikv_client_go_v2//kv",
         "@com_github_tikv_client_go_v2//tikv",
diff --git a/pkg/table/context/BUILD.bazel b/pkg/table/context/BUILD.bazel
index 5eb49d4315..00e6b1929c 100644
--- a/pkg/table/context/BUILD.bazel
+++ b/pkg/table/context/BUILD.bazel
@@ -23,7 +23,6 @@ go_library(
         "//pkg/util/intest",
         "//pkg/util/rowcodec",
         "//pkg/util/tableutil",
-        "@com_github_pingcap_tipb//go-binlog",
     ],
 )
 
diff --git a/pkg/table/contextimpl/BUILD.bazel b/pkg/table/contextimpl/BUILD.bazel
index da643f143c..e1d224c340 100644
--- a/pkg/table/contextimpl/BUILD.bazel
+++ b/pkg/table/contextimpl/BUILD.bazel
@@ -15,8 +15,6 @@ go_library(
         "//pkg/sessionctx/variable",
         "//pkg/table/context",
         "//pkg/util/intest",
-        "@com_github_pingcap_failpoint//:failpoint",
-        "@com_github_pingcap_tipb//go-binlog",
     ],
 )
 
@@ -28,7 +26,6 @@ go_test(
     deps = [
         ":contextimpl",
         "//pkg/meta/model",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/variable",
         "//pkg/table",
         "//pkg/testkit",
diff --git a/pkg/table/tables/BUILD.bazel b/pkg/table/tables/BUILD.bazel
index d9c7aeff2b..156ff70515 100644
--- a/pkg/table/tables/BUILD.bazel
+++ b/pkg/table/tables/BUILD.bazel
@@ -20,7 +20,6 @@ go_library(
         "//pkg/expression/context",
         "//pkg/expression/contextstatic",
         "//pkg/kv",
-        "//pkg/meta",
         "//pkg/meta/autoid",
         "//pkg/meta/model",
         "//pkg/metrics",
@@ -30,7 +29,6 @@ go_library(
         "//pkg/parser/mysql",
         "//pkg/parser/terror",
         "//pkg/sessionctx",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/variable",
         "//pkg/statistics",
         "//pkg/table",
@@ -56,7 +54,6 @@ go_library(
         "@com_github_pingcap_errors//:errors",
         "@com_github_pingcap_failpoint//:failpoint",
         "@com_github_pingcap_log//:log",
-        "@com_github_pingcap_tipb//go-binlog",
         "@com_github_pingcap_tipb//go-tipb",
         "@com_github_tikv_client_go_v2//oracle",
         "@com_github_tikv_client_go_v2//tikv",
diff --git a/tests/realtikvtest/brietest/BUILD.bazel b/tests/realtikvtest/brietest/BUILD.bazel
index 70f784fb3d..c7e6a829fc 100644
--- a/tests/realtikvtest/brietest/BUILD.bazel
+++ b/tests/realtikvtest/brietest/BUILD.bazel
@@ -19,7 +19,6 @@ go_test(
         "//pkg/executor",
         "//pkg/parser/mysql",
         "//pkg/session",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/store/mockstore/mockcopr",
         "//pkg/testkit",
         "//pkg/testkit/testsetup",
diff --git a/tests/realtikvtest/pipelineddmltest/BUILD.bazel b/tests/realtikvtest/pipelineddmltest/BUILD.bazel
index 18c9dd1b21..c3fa442b04 100644
--- a/tests/realtikvtest/pipelineddmltest/BUILD.bazel
+++ b/tests/realtikvtest/pipelineddmltest/BUILD.bazel
@@ -12,7 +12,6 @@ go_test(
     deps = [
         "//pkg/config",
         "//pkg/kv",
-        "//pkg/sessionctx/binloginfo",
         "//pkg/sessionctx/variable",
         "//pkg/testkit",
         "//tests/realtikvtest",

```

**Comment by fzzf678** [comment]
It seems that we also need a PR to modify some mysql test cases.
Form mysql test:
```
time="2024-09-11T10:10:03+08:00" level=error msg="run test [index_merge_delete] err: sql:SET @save_log_bin= @@sql_log_bin;: run \"SET @save_log_bin= @@sql_log_bin;\" at line 219 err Error 1193 (HY000): Unknown system variable 'sql_log_bin'"
```

**Comment by Benjamin2037** [comment]
> It seems that we also need a PR to modify some mysql test cases. Form mysql test:
> 
> ```
> time="2024-09-11T10:10:03+08:00" level=error msg="run test [index_merge_delete] err: sql:SET @save_log_bin= @@sql_log_bin;: run \"SET @save_log_bin= @@sql_log_bin;\" at line 219 err Error 1193 (HY000): Unknown system variable 'sql_log_bin'"
> ```

 https://github.com/PingCAP-QE/tidb-test/pull/2385

File: pkg/sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -3467,10 +3465,6 @@ const (
 	InnodbFastShutdown = "innodb_fast_shutdown"
 	// InnodbLockWaitTimeout is the name for 'innodb_lock_wait_timeout' system variable.
 	InnodbLockWaitTimeout = "innodb_lock_wait_timeout"
-	// SQLLogBin is the name for 'sql_log_bin' system variable.
```
Comment (lance6716) [inline pkg/sessionctx/variable/sysvar.go:3470]: Maybe involve PM to decide if we should keep some level of compatibility with MySQL.

For `sql_log_bin`, we can't implement it in TiDB, so we may allow read it and forbid writing to it. For `log_bin`, seems this is a read-only variable in MySQL, we can also let TiDB always return `OFF` for it.

https://dev.mysql.com/doc/refman/8.4/en/replication-options-binary-log.html#sysvar_sql_log_bin

https://dev.mysql.com/doc/refman/8.4/en/replication-options-binary-log.html#sysvar_log_bin

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

File: pkg/sessionctx/variable/sysvar.go (hunk 1 above)
Comment (Frank945946) [reply pkg/sessionctx/variable/sysvar.go:3470]: TiDB has discarded binlog, which makes it inherently incompatible with MySQL in this aspect. Therefore, keeping the binlog-related parameters is also pointless. It is recommended to remove them as well.

**Review by Frank945946 — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Comment by gengliqi** [comment]
When can this PR be merged? I want to update tipb but found that the compiler will report an error because `binlog` part in tipb has been removed.

**Review by yudongusa — APPROVED** [review]
LGTM after confirmation with the team on the potential business impact mitigation.

**Comment by lance6716** [comment]
```diff
--- pkg/table/tblsession/table_test.go
+++ pkg/table/tblsession/table_test.go
@@ -21,7 +21,6 @@
 	"github.com/pingcap/tidb/pkg/meta/model"
 	"github.com/pingcap/tidb/pkg/sessionctx/variable"
 	"github.com/pingcap/tidb/pkg/table"
-
 	_ "github.com/pingcap/tidb/pkg/table/tables"
 	"github.com/pingcap/tidb/pkg/table/tblsession"
 	"github.com/pingcap/tidb/pkg/util/mock"
```

File: pkg/table/tblsession/table_test.go [hunk 2]
```diff
@@ -17,12 +17,13 @@ package tblsession_test
 import (
 	"testing"
 
+	_ "github.com/pingcap/tidb/pkg/infoschema"
 	"github.com/pingcap/tidb/pkg/meta/model"
-	"github.com/pingcap/tidb/pkg/sessionctx/binloginfo"
 	"github.com/pingcap/tidb/pkg/sessionctx/variable"
 	"github.com/pingcap/tidb/pkg/table"
+
```
Comment (lance6716) [inline pkg/table/tblsession/table_test.go:24]:
```suggestion
```

**Pull request merged**